src/main/java/net/openhft/chronicle/hash/impl/VanillaChronicleHash.java

```diff
@@ -944,6 +949,36 @@ private void mapTierBulksMapped(int upToBulkIndex) throws IOException {
                 firstBulkToMapOffsetWithinMapping);
     }
 
+    private static class PosixFallocateProxy {
```

Comment:
Please extract this as a standalone class. Call it just `PosixFallocate` for consistency (see `PosixMsync` and `WindowsMsync` classes).

Reply:
Code refactored. Any other problems?